### PR TITLE
Change to set st_message payload pointers to null if st_message data

### PR DESCRIPTION
**Pull request by lwilkovich**
*Description of changes:*
streamId -> stream_id
Francis asked that I set aws_iot_st_msg->payload to NULL when the mesage didn't contain payload data

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: source/serializer.c [hunk 1]
```diff
@@ -214,8 +215,15 @@ int aws_iot_st_msg_deserialize_from_cursor(
...
             }
             aws_byte_cursor_advance(cursor, length);
+            payload_check = 1;
         }
     }
+    if (payload_check == 0) {
+        message->payload.allocator = NULL;
+        message->payload.buffer = NULL;
+        message->payload.capacity = 0;
+        message->payload.len = 0;
```
Comment (fufranci) [inline source/serializer.c:225]:
```
AWS_ZERO_STRUCT(*message->payload);
```

Comment (lwilkovich) [reply source/serializer.c:225]: I don't this would work because NULL and 0 are different

Comment (fufranci) [reply source/serializer.c:225]: I am pretty sure they are the same. But it's not a big deal.

**Review by fufranci — DISMISSED** [review]

File: source/serializer.c [hunk 2]
```diff
@@ -214,8 +215,15 @@ int aws_iot_st_msg_deserialize_from_cursor(
                 goto cleanup;
             }
             aws_byte_cursor_advance(cursor, length);
+            payload_check = 1;
         }
     }
+    if (payload_check == 0) {
+        message->payload.allocator = NULL;
```
Comment (DavidOgunsAWS) [inline source/serializer.c:222]: Logic here may be OK to just zero memory on the entire struct.

Comment (lwilkovich) [reply source/serializer.c]: done

File: source/serializer.c [hunk 3]
```diff
@@ -175,19 +175,20 @@ int aws_iot_st_msg_deserialize_from_cursor(
     uint8_t wire_type;
     uint8_t field_number;
     int length;
+    int payload_check = 0;
```
Comment (fufranci) [inline source/serializer.c:178]: `payload_check` should be a `bool`.

**Review by fufranci — APPROVED** [review]

**Pull request merged**